tests/uptane_vector_tests.cc

```diff
@@ -45,67 +45,100 @@ class Uptane_Vector_Test {
+
+    auto storage = INvStorage::newStorage(config.storage);
+    Uptane::Manifest uptane_manifest{config, storage};
+    auto uptane_client = SotaUptaneClient::newDefaultClient(config, storage);
+    Uptane::EcuSerial ecu_serial(config.provision.primary_ecu_serial);
+    Uptane::HardwareIdentifier hw_id(config.provision.primary_ecu_hardware_id);
+    uptane_client->hw_ids.insert(std::make_pair(ecu_serial, hw_id));
+    uptane_client->installed_images[ecu_serial] = "test_filename";
     logger_set_threshold(boost::log::trivial::trace);
 
-    try {
-      auto storage = INvStorage::newStorage(config.storage);
-      Uptane::Manifest uptane_manifest{config, storage};
-      auto uptane_client = SotaUptaneClient::newDefaultClient(config, storage);
-      Uptane::EcuSerial ecu_serial(config.provision.primary_ecu_serial);
-      Uptane::HardwareIdentifier hw_id(config.provision.primary_ecu_hardware_id);
-      uptane_client->hw_ids.insert(std::make_pair(ecu_serial, hw_id));
-      uptane_client->installed_images[ecu_serial] = "test_filename";
-      if (!uptane_client->uptaneIteration()) {
-        throw uptane_client->getLastException();
+    HttpClient http_client;
+    while (true) {
+      HttpResponse response = http_client.post(address + test_name + "/step", Json::Value());
+      if (response.http_status_code == 204) {
+        return true;
       }
+      const auto vector(response.getJson());
+      std::cout << "VECTOR: " << vector;
 
-    } catch (const Uptane::Exception& e) {
-      return match_error(vector, e);
-    } catch (const std::exception& e) {
-      std::cout << "aktualizr failed with unrecognized exception " << typeid(e).name() << ": " << e.what() << "\n";
-      return false;
-    }
+      bool should_fail = false;
+      if (vector["director"]["update"]["is_success"].asBool() == false ||
+          vector["image_repo"]["update"]["is_success"].asBool() == false) {
+        should_fail = true;
+      } else {
+        for (const auto& t : vector["director"]["targets"]) {
+          if (t["is_success"].asBool() == false) {
+            should_fail = true;
+            break;
+          }
+        }
+        for (const auto& t : vector["image_repo"]["targets"]) {
+          if (t["is_success"].asBool() == false) {
+            should_fail = true;
+            break;
+          }
+        }
+      }
 
-    if (vector["director"]["update"]["is_success"].asBool() && vector["image_repo"]["update"]["is_success"].asBool()) {
-      for (Json::ValueConstIterator it = vector["director"]["targets"].begin();
-           it != vector["director"]["targets"].end(); ++it) {
-        if (!(*it)["is_success"].asBool()) {
-          std::cout << "aktualizr did not fail as expected.\n";
+      try {
+        if (!uptane_client->uptaneIteration()) {
+          if (should_fail) {
+            throw uptane_client->getLastException();
+          }
+          std::cout << "aktualizr return error but should not\n";
```

Comment:
Sorry to be pedantic, but just to make this extra clear: "uptaneIteration failed but should have succeeded".